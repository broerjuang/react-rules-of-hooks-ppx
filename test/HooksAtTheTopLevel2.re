[@react.component]
let make = () => {
  let (state, setState) = React.useState();

  <div onClick={_evt => useMouseHook()} />;
};
