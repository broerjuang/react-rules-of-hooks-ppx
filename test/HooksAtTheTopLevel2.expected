File "HooksAtTheTopLevel2.ml", line 3, characters 30-45:
3 |   ((div ~onClick:(fun _evt -> useMouseHook ()) ~children:[] ())[@JSX ])
                                  ^^^^^^^^^^^^^^^
Error: Hooks can't be inside conditionals, neither loops.
